src/components/DateInput/YearAndMonthSelector.tsx

```diff
@@ -1,30 +1,54 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import type { LocaleUtils } from 'react-day-picker';
+
+// { date, localeUtils, onChange, startYear, endYear }
+export interface YearAndMonthSelectorProps {
+  /** Selected or current date as native JS `Date` object */
+  date: Date;
+
+  /** `react-day-picker` localeUtils */
+  localeUtils: LocaleUtils;
+
+  /** Year range start as number */
+  startYear: number;
+
+  /** Year range end as number */
+  endYear: number;
+
+  /** Month/year change callback */
+  onChange?: (date: Date) => void;
+}
 
 /**
  * Private component for DateInput that renders Month/Year select inputs.
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const YearAndMonthSelector = ({ date, localeUtils, onChange, startYear, endYear }) => {
-  const months = localeUtils.getMonths();
+const YearAndMonthSelector = ({
+  date,
+  localeUtils,
+  onChange,
+  startYear,
+  endYear,
+}: YearAndMonthSelectorProps): JSX.Element => {
+  const months: ReturnType<typeof localeUtils.getMonths> = localeUtils.getMonths();
```

Comment:
I'm not this far with TypeScript. Can I understand a bit more of what this is doing?

Reply:
`ReturnType` gives you back the return type of a function type, it's why i did `ReturnType<typeof localeUtils.getMonths> ` rather than `ReturnType<localeUtils.getMonths>, in this case i used it so i wouldn't type tuple of 12 strings manually.

if you do `type Example = ReturnType<() => string>`, you get back `string` as a type for `Example`. in this example, what's inside the angle brackets is also a type, rather than a function reference.

[docs](https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype)

Reply:
oh! That's awesome!